Analysis/src/GenParticle.cpp

```diff
@@ -147,16 +152,15 @@ void GenEvent::PrintChain(const GenParticle* particle, const std::string& pre) c
     const auto particleName = GetParticleName(pdgParticle);
     const int particleStatus = particle->status;
     const LorentzVectorM genParticle_momentum = particle->momentum;
-    const std::bitset<15> genStatusFlags_ = particle->genStatusFlags.flags ;
-    // for (unsigned n = 0; n < iteration; ++n)
-    //     std::cout << "  ";
+    const auto flag = particle->genStatusFlags.GenStatusFlags::getFlags();
+    const std::bitset<15> genStatusFlags_ = flag ;
     auto mothers_index = particle->mothers.size() > 0 ?  particle->mothers.at(0)->index : 0;
     std::cout << particleName                              << " <" << pdgParticle
               << "> pt=" << genParticle_momentum.Pt()      << " eta=" << genParticle_momentum.Eta()
               << " phi=" << genParticle_momentum.Phi()     << " E=" << genParticle_momentum.E()
-              << " m=" << genParticle_momentum.M()         << " index=" << particle->index
-              << " mother_index=" << mothers_index;
+              << " m=" << genParticle_momentum.M()         << " index=" << particle->index;
     if(particle->mothers.size() > 0){
+        std::cout  << " mother_index=" << mothers_index;
```

Comment:
Don't need to do for. Use `CollectionToString` defined here https://github.com/hh-italian-group/AnalysisTools/blob/master/Core/include/TextIO.h#L78
I.e.:
```
std::cout << " mother_index=" << CollectionToString(particle->mothers, ",");
```

Reply:
But in this way, the output of CollectionToString would be each of the GenParticle mothers not their indexes, no?

Reply:
Hm... right. Ok, keep it as it is, but put the definition of `auto mothers_index` inside the if.